### DISCUSSION TITLE
plain/text log healthchecks

### DISCUSSION THREAD

**Discussion by scialo** [garethgeorge/backrest#540] [Ideas]
is it possible send log when run hook healthchecks?
thanks

**Comment by Hoempi** [comment]
Yes, I'm doing it likes this:
![grafik](https://github.com/user-attachments/assets/c754455d-34d3-4047-bee2-f7d9907d3fe7)

The result in my self-hosted healthcheck.io is the following:

`{"text":"\nBackrest Snapshot Notification\nTask: backup for plan \"main\" at 2024-11-05T20:00:02+01:00\nEvent: snapshot success\nSnapshot: 182cfa8d3e197f05acf1f6b9b772f6a4f0e8649d6fc2801478a78390a6841547\nOverview:\n- Data added: 1.001 MB\n- Total files processed: 547\n- Total bytes processed: 11.011 MB\n\nBackup Statistics:\n- Files new: 1\n- Files changed: 5\n- Files unmodified: 541\n- Dirs new: 0\n- Dirs changed: 7\n- Dirs unmodified: 171\n- Data blobs: 6\n- Tree blobs: 8\n- Total duration: 0.85105408s\n\n"}`

**Comment by scialo** [comment]
Currently, when the ping is sent, the content-type is set to ‘application/json’ whereas I am asking if it is possible to have it in ‘text/plain’ as well.

**Comment by inode64** [comment]
When you make the improvement, I did not pay attention to this detail, 
Would a condition be okay to display messages in json, html or text?

**Reply by garethgeorge** [reply]
Thanks for giving a bit more thought for this, overall there's been quite a bit of positive feedback on the addition of the healthchecks.io hook / the convenience of how it automatically picks the right status URL suffix to ping.

Re: the configurability, I'm not sure I understand the usecase for the json formatted payload type? How does that look when it shows up in the healthchecks.io UI and would `text/plain` be a better default?

For context to others, inode64 PR'd the original integration of healthchecks.
